addons/account/models/account_payment_method.py

```diff
@@ -0,0 +1,102 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models
+
+
+class AccountPaymentMethod(models.Model):
+    _name = "account.payment.method"
+    _description = "Payment Methods"
+
+    name = fields.Char(translate=True, required=True)
+    code = fields.Char(required=True)  # For internal identification
+    # category = fields.Selection(selection=[('manual', 'Manual')], required=True)
+    payment_type = fields.Selection(selection=[('inbound', 'Inbound'), ('outbound', 'Outbound')], required=True)
+    # journal_type = fields.Selection(
+    #     selection=[('bank', 'Bank'), ('cash', 'Cash')],
+    #     help='Choose on which journal type this method will be usable. Leave empty to allow both types.',
+    #     required=False
+    # )
+
+    _sql_constraints = [
+        ('name_code_unique', 'unique (code, payment_type)', 'The combination code/payment type already exists!'),
+    ]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        payment_methods = super().create(vals_list)
+        methods_info = self._get_method_information()
+        for method in payment_methods:
+            information = methods_info.get(method.code)
+            if information:
```

Comment:
I think we should force anyone to provide method_information for the newly created payment_method